### DISCUSSION TITLE
Creating(User Registration) users from a frontend form

### DISCUSSION THREAD

**Discussion by supermx1** [Budibase/budibase#1253] [Help] [answered]
Hello, I have been stuck on this part for some hours now. I'm trying to create an app with user registration. Now the login screen is auto-generated in the public part of the app, which is fine, but then I also need a user registration form on the public part of the app. A user who clicks a Create New Account button is redirected to another page. Now I created the page, only to realize that I cannot link it to the Users table because the password field is hidden. This brings me to the next issue, which is I cannot seem to create a password field in the DB. Maybe because of my limited web dev knowledge I cannot seem to figure this out. The documentation does address this either. So how does one do all this?

**Comment by KernOil** [comment]
having the same problem. I need to be able to reset passwords and i was thinking along your same lines. but the passwords fields is hidden.

**Reply by mjashanks** [reply]
Hey Andrew. Next weeks release comes with a prepackaged user Invite flow, and password reset flow.

It will also come will OAuth via google, and we will be adding more providers soon.

**Comment by mjashanks** [comment, marked as answer]
Hey @supermx1. Apologies for not replying to this - it got lost.

As per my comment above, the next release will support a whole user onboarding and management flow.

However, we will not be supporting "Sign Up" right now, as we feel like it does not really make sense for people using Budibase to build apps for their own organisation. Signup feels more like "Signup to use our product". Do you see a use for signup within an organisation?

Apologies again for taking several months to reply!

**Reply by salmanin** [reply]
I understand budibase is prioritizing in-organization business apps, but it would be great to have a simple external "signup" process to have things like tracking, or to show status to an external user (even if it's just something flagged manually by internal users). Sorry if this is already available or part of the onboarding/management flow, I am still exploring all the brilliant features of budibase! :)

**Reply by orenka** [reply]
Is this still the position of Budibase?
I want to build an app that  allows only registered users to use, and they are not in my organization (I don't even have an organization - I just want to build a simple dara-oriented app for registered users).
Note that currently I cannot even add user manually, as the "add row" button does not appear in the context of the "Users" table.

Also, even for users within the organization – how would such users be added to the Users table? They still need to be registered and I couldn't even login (using a different user ID) to an app that I published.

Is there a guide on how multiple users can access the same app and register themselves on the app's Users table? the way BB is setup now it is not clear how to use the Users table.
Is there a way to implement my plan to allow any users to login?
thank you.

**Reply by ravishk** [reply]
+1 on orenka's q.

I recently started evaluating Budibase (& so far have been blown away by its features & simplicity) for a potential use case to build & launch simple data-oriented apps for some of our customers. This gives customers a view of some of the services we are performing for them. IMO, BB fits the bill perfectly for this use case. 

Thanks!

**Reply by RASS00** [reply]
@orenka @ravishk pretty much yes, you still need to run an SSO system to redirect them but if you use the api, it works flawlessly, https://docs.budibase.com/reference/post_users
there's a few things that you need to configure but ultimately it's nothing.

**Reply by gaa23** [reply]
I am also interested in using Budibase to build apps for external clients and would like to have access to related features.

**Reply by marian001** [reply]
+1

I understand API can be used to create a user, but I haven't seen an API call which would also assign a user to an application. Is that possible any other way than manually in application management?

UPDATE: Found it..   So I guess the last thing I'm struggling with is how to query the Main User Table (i.e. user table outside of specific application).  How can it be done?

**Reply by chadgriff** [reply]
@marian001 I am also looking for the method wherein I can use an API call to assign a user to an application. Can you point me towards what you found?

**Reply by marian001** [reply]
Hi @chadgriff ,
I'm first getting list of all users (using this API call: https://docs.budibase.com/reference/post_users-search) and then filtering those, who don't have any role assigned for my app. I then update each of those users (using this API call: https://docs.budibase.com/reference/put_users-userid) by assigning them BASIC role for my app. This runs every minute and also generates email notification to the user..
